### ISSUE TITLE
Customization?

### ISSUE THREAD

**Issue by Lewenhaupt** [rbreaves/kinto#643]
Hi,

When I use mac I usually swap cmd and ctrl such that I do copy and paste as I would on a windows/linux lapop, i.e. ctrl+c and ctrl+v, however I tend to like macs way of doing @ (opt + 2 for a swedish keyboard, compared to altgr + 2 on windows) but it seems I can only have the cmd+c cmd+v and the altgr +2 out of the box. I'm not sure if there is any easy way of accomplishing this, because I really love how this app simply solved all my issues with different terminals etc. having different ways of copying/pasting etc.

**Comment by rbreaves** [comment]
Sure, I do not think it'd be too awfully hard, I believe the readme delves a bit into how you can modify it for Linux or Windows after the install. (backup that file if you ever re-install though as I do not yet ensure to keep modifications you make on re-installs or updates)

Linux
~/.config/kinto/kinto.py

Windows
~\.kinto\kinto.ahk

In both cases things are blocked off to remap modifier keys first, the big 3 - ctrl, alt and win/super and then the rest is more specific to types of applications, terminals vs GUI apps, text editors, remote apps, etc.

**Comment by Lewenhaupt** [comment]
Thanks for the quick reply. I gave it a go and in terminal I finally found some settings that I think gave me what I wanted. However, I quickly encountered issues when I went into Webstorm. I think this might just be because I want LEFT_ALT to act as RIGHT_CTRL as it seems to make most commands make sense but I don't think Webstorm recognizes a difference in LEFT_CTRL and RIGHT_CTRL. So I think I'll have to dig into actually changing each command I use in IntelliJ stuff specifically for that to work.

But I encountered another think, which I'm a bit curious about. I can't see in the config how you manage to accomplish the alt+c/alt+v copy/paste in the terminals? I can't seems to find it explicitly anywhere and thus I couldn't find a way to skip any remappings other than those.

**Comment by rbreaves** [comment]
> I can't see in the config how you manage to accomplish the alt+c/alt+v copy/paste in the terminals?

It's a layered approach, starting on like 110.

https://github.com/rbreaves/kinto/blob/master/linux/kinto.py#L110
If your are not in the terminal group then you don't get the terminal modifier key remaps - hence the GUI app group. All things not in the terminal group is then GUI by default.
```
define_conditional_modmap(lambda wm_class: wm_class.casefold() not in terminals,{
```

If you are a terminal app which is defined further near the top then like 145 handles remapping those terminals.
https://github.com/rbreaves/kinto/blob/master/linux/kinto.py#L145
```
define_conditional_modmap(re.compile(termStr, re.IGNORECASE), {
```

This concept is pretty well what is at the core of Kinto.sh and why it works as well as it does. I am not trying to remap everything individually at all. I just group things together when I can - and when I can't I then tackle those individually. Sadly many people act as though "they can catch'em all" or some crazy thing like that... I don't know about you but I have better things to do with my time than to create new entries all the time for some new random app I might be using lol.

I use what I use and I approximate the rest, when others contribute more for specific apps or behavior I accept them. Keeps things simple.

**Comment by Lewenhaupt** [comment]
I must have done something wrong then. I commented all the mappings in both general and terminal mappings and it still allowed alt+c for copy. I'll give it another go and see if I can figure out what went wrong.

**Comment by rbreaves** [comment]
You may be thinking of the other specific code block just for terminals at the bottom of the file or you missed that - those are specific key remaps not the general modifier key remap only.

Like I said it’s all done in layers & builds.

**Comment by Lewenhaupt** [comment]
Yeah restarting I got 😊 might be that I missed that section then, thanks for pointing it out! 👍

**Comment by rbreaves** [comment]
Skimming back through my readme I had some really old info in there about the prior version of Kinto that used a tmp folder location.. that is now taken out. I appeared to say it ran real time too, I did not recall that, but regardless it is updated now.

I do think there is a watch param for xkeysnail but maybe it doesn't or didn't actually work. I always have to restart the app after making a change. Closing ticket.
